pkgs/development/compilers/zig/generic.nix

```diff
@@ -0,0 +1,67 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, cmake
+, llvmPackages
+, libxml2
+, zlib
+, coreutils
+, ...
+}:
+
+hash: args:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "zig";
+
+  src = fetchFromGitHub {
+    owner = "ziglang";
+    repo = "zig";
+    rev = finalAttrs.version;
+    inherit hash;
+  };
+
+  nativeBuildInputs = [
+    cmake
+    llvmPackages.llvm.dev
+  ];
+
+  buildInputs = [
+    libxml2
+    zlib
+  ] ++ (with llvmPackages; [
+    libclang
+    lld
+    llvm
+  ]);
+
+  doInstallCheck = true;
+
+  # Zig's build looks at /usr/bin/env to find dynamic linking info. This doesn't
+  # work in Nix's sandbox. Use env from our coreutils instead.
+  postPatch = ''
+    substituteInPlace lib/std/zig/system/NativeTargetInfo.zig \
+      --replace "/usr/bin/env" "${coreutils}/bin/env"
+  '';
+
+  preBuild = ''
+    export ZIG_GLOBAL_CACHE_DIR=$(mktemp -d)
+  '';
+
+  installCheckPhase = ''
```

Comment:
```suggestion
  doInstallCheck = true;
  installCheckPhase = ''
```

Reply:
I'm following `nixpkgs-hammering`'s convention: https://github.com/jtojnar/nixpkgs-hammering/blob/243b81c687aac33d6716957c0cd2235c81631044/explanations/attribute-ordering.md

Reply:
It makes no sense to me to keep `doPhase` and `phase` too far.